windows/src/desktop/kmshell/install/Keyman.Configuration.System.TIPMaintenance.pas

```diff
@@ -62,7 +63,43 @@ implementation
   utilexecute,
   utilsystem;
 
-{ TTIPMaintenance }
+  { For temporary debugging couldn't include utilkeyman.pas so extracted needed function }
+
+function GetShortKeyboardName(const FileName: string): string;
+begin
+  if (LowerCase(ExtractFileExt(FileName)) = '.kmx') or
+      (LowerCase(ExtractFileExt(FileName)) = '.kxx') or
+      (LowerCase(ExtractFileExt(FileName)) = '.kmp')
+    then Result := ChangeFileExt(ExtractFileName(FileName), '')
+    else Result := FileName;
+end;
+
+// https://stackoverflow.com/questions/9004870/enumerate-registry-subkeys-in-delphi
+procedure GetSubKeyList(RootKey: HKEY; const Key: string; var KeyList: string);
+var
+  Registry: TRegistry;
+  SubKeyNames: TStringList;
+  Name: string;
+begin
+  Registry := TRegistry.Create(KEY_READ);
+  KeyList := '';
+  Try
+    Registry.RootKey := RootKey;
+    Registry.OpenKeyReadOnly(Key);
```

Comment:
```suggestion
    if not Registry.OpenKeyReadOnly(Key) then
    begin
      KeyList := '<Unable to open key>';
      Exit;
    end;
```

Reply:
I did make this change just not using the commit suggestion